### PR TITLE
Fix tutorial rendering. 

### LINKED ISSUE #2189: Tutorial - instruction clarification
Step 5 on the Getting Started Tutorial - add "in an On Shake block" to the instruction text.

gettingstarted

### PR DESCRIPTION
**Pull request by microsoftsam**
Fix bug in tutorial rendering logic. Fixes #2189

The issue was after the full page was rendered by markdown, I had HTML in string format, and I needed the first element of each step. I was using regex to parse the HTML. I am now using the contentdiv passed in and temporarily setting the HTML of each step, and extracting the first elements HTML. Using the DOM to do the parsing.

### CODE REVIEW

**Comment by microsoftsam** [comment]
Needs to be cherrypicked into v0

File: pxtrunner/runner.ts [hunk 1]
```diff
@@ -596,9 +596,11 @@ ${files["main.ts"]}
                         // Split the steps
                         let stepcontent = content.innerHTML.split(/<h3.*\/h3>/gi);
                         for (let i = 0; i < stepcontent.length - 1; i++) {
-                            stepInfo[i].headerContent = stepcontent[i + 1].split(/(<.*?>.*<\/.*?>)/i)[1];
+                            content.innerHTML = stepcontent[i + 1];
```
Comment (pelikhan) [inline pxtrunner/runner.ts]: Isn't there a dom parser to do this kind of stuff?

Comment (microsoftsam) [inline pxtrunner/runner.ts]: There is most definitely a library, but I don't see a need to add another dependency. 
I have figured out a way to use the content element (ie: the passed in DOM) to do the parsing.

**Comment by pelikhan** [comment]
The browser has a DOMParser right? Does your trick influence the HTML layout?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by microsoftsam** [comment]
Nope my trick doesn't influence the HTML layout, as in the case of the tutorial project the content div is never returned, instead the content is returned in messages. The tutorial iframe is 1px by 1px iframe.
Yes I'm using the browser's DOM parser with this.

**Comment by microsoftsam** [comment]
This needs to be tested on firefox, etc. but I think the reason this trick works is because the content div is already added to the tutorial iframe's DOM. Otherwise you get weird rendering issues, because the font sizes and such aren't passed in.

**Comment by pelikhan** [comment]
Sounds good then.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by pelikhan** [comment]
Oh font size computation. Makes sense now. Yes it has to be in the DOM.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by microsoftsam** [comment]
Fails in FF, back to the drawing board.

**Comment by microsoftsam** [comment]
Fixed in FF.

**Comment by microsoftsam** [comment]
Need to test on IE / Edge and Windows.
